### PR TITLE
[bugfix] Fix call rate in sample_qc to divide by the total rows.

### LINKED ISSUE #5561: call_rate incorrect in sample_qc
In `sample_qc`, `call_rate` is computed as `n_called / (n_called + n_not_called)`, where `n_called` is computed as
```
hl.agg.count_where(hl.is_defined(mt['GT']))
```
and `n_not_called` is
```
hl.agg.count_where(hl.is_missing(mt['GT']))
```

But in `variant_qc`, `n_not_called` is `n_samples - n_called`, and `call_rate` is `n_called / n_samples`.

After a `filter_entries`, these two methods give different answers (the first is the call rate as a fraction of unfiltered entries, while the second is as a fraction of all samples). I believe `variant_qc` is correct here, but whatever is correct, `variant_qc` and `sample_qc` need to be consistent with each other.

[Zulip discussion](https://hail.zulipchat.com/#narrow/stream/123010-Hail-0.2E2.20support/topic/Call.20rate.20calculated.20by.20sample_qc.20is.20counterintuitive/near/160264785)

### PR DESCRIPTION
**Pull request by tpoterba**
Fixes #5561

Also improved the implementation of sample and variant qc.

### CODE REVIEW

File: hail/python/test/hail/methods/test_qc.py [hunk 1]
```diff
@@ -174,3 +174,14 @@ def test_filter_alleles_hts(self):
...
         )
+
+    def test_sample_and_variant_qc_call_rate(self):
+        mt = hl.import_vcf(resource('sample.vcf'))
+
+        n_rows, n_cols = mt.count()
+        mt = mt.filter_entries(mt.GQ > 5)
+        mt = hl.variant_qc(hl.sample_qc(mt))
+
+        mt.select_cols(mt.sample_qc.call_rate, foo = mt.sample_qc.n_called / n_rows).cols().show()
```
Comment (jigold) [inline hail/python/test/hail/methods/test_qc.py]: Delete? Not asserting anything?

File: hail/python/hail/methods/qc.py [hunk 2]
```diff
@@ -244,51 +242,47 @@ def variant_qc(mt, name='variant_qc') -> MatrixTable:
...
+                                                                e1.call_stats.AC[1] - 2 *
+                                                                e1.call_stats.homozygote_count[1],
+                                                                e1.call_stats.homozygote_count[1])
+                                         ).or_missing(),
+                          lambda hwe: hl.struct(**{
+                              **gq_dp_exprs,
+                              **e1.call_stats,
+                              'call_rate': hl.float(e1.n_called) / (e1.n_called + e1.n_not_called + e1.n_filtered),
+                              'n_called': e1.n_called,
+                              'n_not_called': e1.n_not_called,
```
Comment (jigold) [inline hail/python/hail/methods/qc.py]: does the order of the fields matter for backwards compatibility?

File: hail/python/test/hail/methods/test_qc.py (hunk 1 above)
Comment (tpoterba) [reply hail/python/test/hail/methods/test_qc.py]: oops

File: hail/python/hail/methods/qc.py (hunk 2 above)
Comment (tpoterba) [reply hail/python/hail/methods/qc.py]: Really any change in schemas could technically break a pipeline, I suppose. I'm comfortable with this, though.

**Pull request merged**